exporters/dynamicconfigloader/dynamicconfigloader.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamicconfigloader
```

Comment:
I feel "loader" sounds like a one-time tool that load config from some backends. A "watcher" is probably closer to the meaning of watch config changes. This module can define

MetricConfig struct { Period time.Duration; }
ConfigWatcher interface {
OnInitialConfig(MetricConfig);
OnUpdatedConfig(MetricConfig);
}

Register(ConfigWatcher watcher)
Unregister(ConfigWatcher watcher)

Then, in controller/push.go, it can implement a config watcher, register the watch on construction, and unregister on destruction.

We can call this module confignotifier, it will have a goroutine to periodically check config changes. If there is a change, it should call the OnUpdatedConfig() for all registered listeners.

Reply:
Also, we will add parts to periodically check config changes, so we can simply name the directory "dynamicconfig".